pkg/edit/insert_api.go

```diff
@@ -8,11 +8,53 @@ import (
 	"github.com/xiaq/persistent/hashmap"
 )
 
+//elvdoc:var abbr
+//
+// This is a map of abbreviations to expansions. The expansion occurs as soon
+// as the final character of the abbreviation is typed. If more than a single
+// abbreviation would match the longest one is used.
+//
+// Examples:
+//
+// ```elvish
+// edit:abbr['||'] = ' | less'
+// edit:abbr['>dn'] = ' 2>/dev/null '
+// ```
+//
+// @cf edit:word-abbr
+
+//elvdoc:var word-abbr
+//
+// This is a map of small word abbreviations to expansions. The behavior
+// mimics that of abbreviations in editors such as Vim and Emacs. Most notably
+// word abbreviations are expanded only when a non-small word character is
+// typed and the character preceding the abbreviation is not a small word
+// character.
+//
+// Note that the beginning and ending of a typed line of text is considered to
+// be a non-small word character. This means that word abbreviations expand
+// when typed at the start of a line and followed by a non-small word char or
+// when [enter] is pressed.
+//
```

Comment:
Document what happens when both small-word abbreviation expansion and (normal) abbreviation expansion are both eligible: the former takes precedence.

Reply:
I've restructured the code so that "regular" abbreviations have higher precedence and documented that fact.